src/Ocelot/Authorization/ClaimsAuthorizer.cs

```diff
@@ -38,7 +37,7 @@ List<PlaceholderNameAndValue> urlPathPlaceholderNameAndValues
                 if (values.Data != null)
                 {
                     // dynamic claim
-                    var match = Regex.Match(required.Value, @"^{(?<variable>.+)}$");
+                    var match = _regexAuthorize.Match(required.Value);
```

Comment:
No catching and processing of the `RegexMatchTimeoutException` exception.

Will it result to unhandled exception by app domain?
Wil it be caught by upper context?